ui/src/services/k8s/solutions.js

```diff
@@ -96,18 +98,57 @@ export async function getEnvironmentAdminUIs(environment) {
   }));
 }
 
-export async function getEnvironmentConfigMap(environment) {
-  for (const namespace of environment.namespaces) {
-    const result = await coreV1.readNamespacedConfigMap(
+export async function getNamespacedConfigmap(environment) {
+  try {
+    return await coreV1.readNamespacedConfigMap(
       ENVIRONMENT_CONFIGMAP_NAME,
-      namespace,
+      environment,
     );
+  } catch (error) {
+    return { error };
+  }
+}
 
-    if (!result.error) {
-      // Return the first one found among the Environment namespaces
-      return { ...result?.data };
-    }
+export async function createNamespacedConfigMap(namespace) {
+  const body = {
+    metadata: {
+      name: ENVIRONMENT_CONFIGMAP_NAME,
+    },
+  };
+  try {
+    return await coreV1.createNamespacedConfigMap(namespace, body);
+  } catch (error) {
+    return { error };
   }
 }
 
+export async function patchNamespacedConfigMap(namespace, data) {
+  const body = {
+    data: { ...data },
+  };
+  try {
+    return await coreV1.patchNamespacedConfigMap(
+      ENVIRONMENT_CONFIGMAP_NAME,
+      namespace,
+      body,
+      undefined,
+      undefined,
+      {
+        headers: {
+          'Content-Type': 'application/merge-patch+json',
+        },
+      },
+    );
+  } catch (error) {
+    return { error };
+  }
+}
+
+export async function getNamespacedDeployment(name, namespace) {
+  try {
+    return await appsV1.readNamespacedDeployment(name, namespace);
+  } catch (error) {
+    return { error };
+  }
+}
```

Comment:
Definitely a helper to put in `core` (or `app`, if you want to add one, but I'd say `core` is for everything "core" K8s, while others are for Metal-specific concepts).